### PR TITLE
Add install instructions to CMakeLists.txt

### PR DESCRIPTION
**Pull request by Jochen0x90h**
also add -fPIC to be able to link cpr into a shared library

### CODE REVIEW

**Comment by tastytea** [comment]
Please use [GNUInstallDirs](https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html). It is portable and the variables can be overridden.

**Comment by jakesyl** [comment]
@tastytea why not just do this in the CMake?

**Comment by tastytea** [comment]
@jakesyl What do you mean? We talk about a CMake file here?

**Comment by COM8** [comment]
Ping @kingkili.

File: cpr/CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,7 @@ add_library(${CPR_LIBRARIES}
     "${CPR_INCLUDE_DIRS}/cpr/ssl_options.h")
 
 message(STATUS "Using CURL_LIBRARIES: ${CURL_LIBRARIES}.")
+set_property(TARGET ${CPR_LIBRARIES} PROPERTY POSITION_INDEPENDENT_CODE ON)
```
Comment (KingKili) [inline cpr/CMakeLists.txt:43]: I think this should depend on BUILD_SHARED_LIBS

File: cpr/CMakeLists.txt [hunk 2]
```diff
@@ -53,3 +54,12 @@ else()
         ${CPR_INCLUDE_DIRS}
         ${CURL_INCLUDE_DIRS})
 endif()
+
+
+# install headers
+install(DIRECTORY "${CPR_INCLUDE_DIRS}/cpr/" DESTINATION include/cpr FILES_MATCHING PATTERN "*.h")
```
Comment (KingKili) [inline cpr/CMakeLists.txt:60]: Isn't this achieved with [this](https://github.com/whoshuu/cpr/blob/00d87d8e447802f88218bf826f4359c9ffeeb80c/CMakeLists.txt#L83)?

File: cpr/CMakeLists.txt [hunk 3]
```diff
@@ -53,3 +54,12 @@ else()
...
+
+# install the library
+install(TARGETS ${CPR_LIBRARIES}
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib)
```
Comment (KingKili) [inline cpr/CMakeLists.txt:65]: This should also be on master

**Comment by COM8** [comment]
With #383 I would say all features proposed here have been merged.
I’m closing this.
Feel free to open a new PR if something is missing.

**Pull request closed without merging**